### ISSUE TITLE
Internal segfault with Zimbra email account

### ISSUE THREAD

**Issue by valmar** [Foundry376/Mailspring#87]
With one IMAP provider, mailspring seems to experience an internal crash. After the crash, a pop up comes up pointing me to "Preferences > Accounts", where a message tells me to remove and readd the account. Sometimes, the pop up does not come up, and mailspring keeps running using high CPU resources.

I am running Mailspring 1.0.2 on Ubuntu Beta 17.10.

The error happens with my employer's email server: mail.desy.de. This is a Zimbra mail server. In my email account I have several (30+) folders.

Here is the error I get in the terminal:

{ Error: SyncWorker crashed with null (code 1):
Error: Account is missing required fields:imap_*
    at EventEmitter._electron.ipcMain.on (/src/browser/application.es6:637:19)
    at emitTwo (events.js:106:13)
    at EventEmitter.emit (events.js:194:7)
    at WebContents.<anonymous> (/usr/share/mailspring/resources/electron.asar/browser/api/web-contents.js:256:13)
    at emitTwo (events.js:106:13)
    at WebContents.emit (events.js:194:7)
  message: 'SyncWorker crashed with null (code 1):\nError: Account is missing required fields:imap_*' } { log: '5984 [2017-10-08 11:33:04.302] [background] [info] 🔸 Updating message 80448=80448 with subject:

[Hidden because it contains private information]

[2017-10-08 11:33:07.566] [metadataExpiration] [info] Scanning for expired metadata\n5984 [2017-10-08 11:33:07.567] [metadataExpiration] [info] -- Will wake for next expiration in 7201sec\n5984 [2017-10-08 11:33:34.105] [background] [critical] \n ***\n *** A C++ exception occurred during program execution: \n *** std::bad_alloc\n ***\n\n5984 [2017-10-08 11:33:34.195] [background] [critical]  *** Stack trace (line numbers are approximate):\n *** (unknown)  clone\n ***\n\n',
  provider: 'imap' }

Please let me know if you need more information. This only happens with this account. My Gmail account works perfectly.

Apart from this, mailspring is going to be my email client of choice (Athough I wish for an icon with rounded borders)

Valerio

**Comment by Bouska** [comment]
I'm running MailSpring **1.0.3** in safe mode (otherwise I go through the onboarding process that crashes see #112) on a **Fedora 26** and I'm getting a similar behavior. I have no problem with a Gmail account but when I add an IMAP account (my uni email account _supelec.fr_, runs on Dovecot I think), the CPU usage of MailSpring goes to the roof. 

Sometimes I get backtrace, sometimes not. The first time I added the account I got this error:
```
{ [Error: SyncWorker crashed with null (code 1)] message: 'SyncWorker crashed with null (code 1)' } { stack: '',
  rawMessage: 'Error: Account is missing required fields:imap_*',
  provider: 'imap',
  log: '' }
```
First time I tried to _Update connection settings..._ of the same account I got the following crash:
```
Waiting for Account JSON:

Waiting for Identity JSON:

 ***
 *** A segmentation fault occurred during program execution.
 *** This typically happens when you try to dereference a pointer
 *** that is NULL or invalid.
 ***
6575 [2017-10-11 04:12:15.979] [main] [critical]  *** Stack trace (line numbers are approximate):
 *** ??:?  runTestAuth(shared_ptr)
 *** ??:?  main
 ***

/usr/share/mailspring/resources/app.asar.unpacked/mailsync: line 4:  6575 Segmentation fault      (core dumped) SASL_PATH="$(dirname $(realpath $0))" LD_LIBRARY_PATH="$(dirname $(realpath $0));$LD_LIBRARY_PATH" "$(dirname $0)/mailsync.bin" "$@"
```
As I'm blocked in safe mode, I can't really be more helpful than that.

**Comment by valmar** [comment]
I still have the same crash in 1.0.5.

I really like Mailspring and I am slowly waiting for it to become more stable!

/usr/share/mailspring/resources/app.asar.unpacked/mailsync: line 4:  4515 Aborted                 (core dumped) SASL_PATH="$(dirname $(realpath $0))" LD_LIBRARY_PATH="$(dirname $(realpath $0));$LD_LIBRARY_PATH" "$(dirname $0)/mailsync.bin" "$@"

 message: 'SyncWorker crashed with null (code 134)' } { stack: '\n\n/usr/share/mailspring/resources/app.asar.unpacked/mailsync: line 4:  4515 Aborted                 (core dumped) SASL_PATH="$(dirname $(realpath $0))" LD_LIBRARY_PATH="$(dirname $(realpath $0));$LD_LIBRARY_PATH" "$(dirname $0)/mailsync.bin" "$@"\n\n',
  rawMessage: 'Error: null4515 [2017-10-17 11:22:57.727] [background] [critical] \n ***\n *** A C++ exception occurred during program execution: \n *** std::bad_alloc\n ***\n\n4515 [2017-10-17 11:22:57.742] [background] [critical]  ',
  provider: 'imap',
  log: '4515 [2017-10-17 11:22:18.148] [main] [info] [local-e02fd392-d2ca] -- Succeeded. Changing status to `remote`\n4515 [2017-10-17 11:22:18.149] [foreground] [info] Idle exited with code 0\n4515 [2017-10-17 11:22:18.150] [foreground] [info] [local-e02fd392-d2ca] Running ChangeUnreadTask performRemote:\n4515 [2017-10-17 11:22:18.163] [foreground] [info] [local-e02fd392-d2ca] -- Succeeded. Changing status to `complete`\n4515 [2017-10-17 11:22:18.164] [foreground] [info] Idling on folder INBOX\n4515 [2017-10-17 11:22:51.742] [metadata] [info] Metadata delta stream closed.\n4515 [2017-10-17 11:22:51.881] [metadata] [info] Encountered error: {"debuginfo":"/delta/streaming","key":"Transferred a partial file","retryable":true,"what":"std::exception"}. . Will retry in 3 sec.\n4515 [2017-10-17 11:22:54.881] [metadata] [info] Metadata delta stream starting...\n4515 [2017-10-17 11:22:57.727] [background] [critical] \n ***\n *** A C++ exception occurred during program execution: \n *** std::bad_alloc\n ***\n\n4515 [2017-10-17 11:22:57.742] [background] [critical]  *** Stack trace (line numbers are approximate):\n *** (unknown)  clone\n ***\n\n' }
Creation of this event was denied due to rate limiting

**Comment by valmar** [comment]
Fixed in 1.0.7 for me

**Comment by letic** [comment]
I can confirm this is solved in 1.0.7 on Linux with a Zimbra server. 

Thanks
